XUtils.c

```diff
@@ -224,6 +232,209 @@ size_t String_safeStrncpy(char* restrict dest, const char* restrict src, size_t
    return i;
 }
 
+#ifdef HAVE_LIBNCURSESW
+ATTR_NONNULL_N(3)
+static size_t String_encodeWChar(void* pos, wchar_t wc, void* state) {
+   char tempBuf[MB_LEN_MAX];
+
+   size_t len = wcrtomb((pos ? (char*)pos : tempBuf), wc, (mbstate_t*)state);
+   assert(len != (size_t)-1);
+   assert(len > 0);
+
+   return len;
+}
+#else
+static size_t String_encodeWChar(void* pos, int c, void* state) {
+   (void)state;
+
+   if (pos) {
+      *(char*)pos = (char)c;
+   }
+   return 1;
+}
+#endif
+
+size_t EncodePrintableString(void* dest, const char* src, size_t srcLen, EncodeWChar encodeWChar, void* encState) {
+   assert(src || srcLen == 0);
+
+   size_t srcPos = 0;
+   size_t destSize = 0;
+   bool wasReplaced = false;
+
+#ifdef HAVE_LIBNCURSESW
+
+   const wchar_t replacementChar = CRT_utf8 ? L'\xFFFD' : L'?';
+
+   mbstate_t decState;
+   memset(&decState, 0, sizeof(decState));
+
+   mbstate_t newState;
+   wchar_t wc;
+   do {
+      wc = L'\0';
```

Comment:
```suggestion
      wc = 0;
```

This makes `c` and `wc` assignments compatible and thus would not require distinction in the code paths …